sympy/core/backend.py

```diff
@@ -0,0 +1,23 @@
+import os
+USE_SYMENGINE = os.getenv('USE_SYMENGINE')
+
+if USE_SYMENGINE:
+    from symengine.sympy_compat import (Symbol, Integer, sympify, S,
+        SympifyError, exp, log, gamma, sqrt, I, E, pi, Matrix,
+        sin, cos, tan, cot, csc, sec, asin, acos, atan, acot, acsc, asec,
+        sinh, cosh, tanh, coth, asinh, acosh, atanh, acoth,
+        lambdify, symarray, diff, zeros, eye, diag, ones, zeros,
+        expand, Function, symbols, var, Add, Mul, Derivative)
```

Comment:
@certik, with the new changes, notice that horrible `AddClass` workarounds are not there anymore.